fsspec/implementations/sftp.py

```diff
@@ -78,34 +78,46 @@ def rmdir(self, path):
         self.ftp.rmdir(path)
 
     def info(self, path):
-        s = self.ftp.stat(path)
-        if S_ISDIR(s.st_mode):
+        stat = self._decode_stat(self.ftp.stat(path))
+        stat["name"] = path + "/" if stat["type"] == "directory" else path
+        return stat
+
+    @staticmethod
+    def _decode_stat(stat, parent_path=None):
+        if S_ISDIR(stat.st_mode):
             t = "directory"
-        elif S_ISLNK(s.st_mode):
+        elif S_ISLNK(stat.st_mode):
             t = "link"
         else:
             t = "file"
-        return {
-            "name": path + "/" if t == "directory" else path,
-            "size": s.st_size,
+        out = {
+            "name": "",
+            "size": stat.st_size,
             "type": t,
-            "uid": s.st_uid,
-            "gid": s.st_gid,
-            "time": s.st_atime,
-            "mtime": s.st_mtime,
+            "uid": stat.st_uid,
+            "gid": stat.st_gid,
+            "time": stat.st_atime,
```

Comment:
Maybe a good time to settle on standard names for these fields, when available - see https://github.com/intake/filesystem_spec/issues/526

Reply:
Good point the one raised in #526. I can do the conversion here already if it is decided.